### ISSUE TITLE
no-undef mistakenly (?) claims 'Int32Array' is not defined

### ISSUE THREAD

**Issue by JobLeonard** [eslint/eslint#7440]
**Tell us about your environment**

* **ESLint Version: v3.8.1**
* **Node Version: v4.6.0**
* **npm Version: 3.10.9**

Selection or rules from `.eslintrc.json`:
```
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module",
        "ecmaFeatures": {
            "impliedStrict": true,
            "jsx": true,
            "experimentalObjectRestSpread": true
        }
    },
    "env": {
        "browser": true
    },
```

**What did you do? What did you expect to happen? What actually happened?**

See screenshot:

![image](https://cloud.githubusercontent.com/assets/259840/19639168/8dcbc924-99d7-11e6-8e8d-ade0f88a48f1.png)

Spotted in [VSCode 1.6.1](https://code.visualstudio.com/updates), using [ESlint plug-in](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint).

Typed arrays are [part of the ES6 standard](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Int32Array), and I've set my environment to browser too, so I think this should not be triggering `no-undef`, right?

I did a quick search, [it seems this issue has been raised and fixed in the past](https://github.com/eslint/eslint/issues/846) so somewhere a regression must have slipped in (or I'm screwing up my ESLint configuration somehow, that's of course also an option).

**Comment by kaicataldo** [comment]
Hey there, thanks for the issue. Unless part of the config above is missing, you'll also need to enable the `es6` env for ES6 globals. Can you try that out and let us know if that fixes your issue?

**Comment by JobLeonard** [comment]
Ah, right, adding this to `env` fixed it:
```
    "env": {
        "es6": true
    },
```

Apologies for the noise, thank you!

**Comment by kaicataldo** [comment]
Happy to help :)
